### PR TITLE
frontend: added enzyme and migrated tests

### PR DESCRIPTION
**Pull request by BrianCraig**
note: this adds @wojtekmaj/enzyme-adapter-react-17 which
is not the official react@17 adapter, because it still is
in development

### CODE REVIEW

**Comment by thisconnect** [comment]
I think it is fine to use enzyme for the existing tests, but for new ones I'd prefer if we use testing library.

https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/#why-should-i-use-the-react-testing-library

Would also like to discuss this with @x1ddos and @benma

**Comment by BrianCraig** [comment]
@thisconnect could you check this PR? the tests are passing, still they output warnings about I18Language acting outside the tests (couldn't catch the `act`), and react warnings.

https://github.com/digitalbitbox/bitbox-wallet-app/runs/4468648188?check_suite_focus=true#step:4:1735

**Comment by thisconnect** [comment]
I also get an error about the Feetarget test, but I don't fully understand it yet. As far as I see, feetarget uses 2 `<Input>` componen, both use value prop with non optional string, why does it complain about that it is uncontrolled 🤔 

```
      Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
          at input
          at div
          at Input (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/src/components/forms/input.tsx:48:5)
          at FeeTargets (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/src/routes/account/send/feetargets.tsx:55:50)
          at WrapperComponent (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/node_modules/@wojtekmaj/enzyme-adapter-utils/src/createMountWrapper.jsx:48:26)

```

**Comment by thisconnect** [comment]
About the warning:
`react-i18next:: It seems you are still using the old wait option, you may migrate to the new useSuspense behaviour.`

I've also seen this in `webdev`, maybe we better upgrade to the newest version of `react-i18next` first?
I hope it still supports class components.

**Comment by thisconnect** [comment]
Also about `act`, I actually haven't seen this api, would it make sense to try using latest i18next / react-i18next first?

In worst case we can skip `language.test.tsx`, this shouldn't be a blocker to migrate to react.

**Comment by BrianCraig** [comment]
> I also get an error about the Feetarget test, but I don't fully understand it yet. As far as I see, feetarget uses 2 `<Input>` componen, both use value prop with non optional string, why does it complain about that it is uncontrolled thinking
> 
> ```
>       Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
>           at input
>           at div
>           at Input (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/src/components/forms/input.tsx:48:5)
>           at FeeTargets (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/src/routes/account/send/feetargets.tsx:55:50)
>           at WrapperComponent (/Users/connect/go/src/github.com/digitalbitbox/bitbox-wallet-app/frontends/web/node_modules/@wojtekmaj/enzyme-adapter-utils/src/createMountWrapper.jsx:48:26)
> ```

That is because Input at https://github.com/digitalbitbox/bitbox-wallet-app/blob/3d8c56a8e84d03259045c3e3c269b2341e5b78a2/frontends/web/src/routes/account/send/feetargets.tsx#L139 is re-rendered with https://github.com/digitalbitbox/bitbox-wallet-app/blob/3d8c56a8e84d03259045c3e3c269b2341e5b78a2/frontends/web/src/routes/account/send/feetargets.tsx#L234 .
The first one doesn't have a value, and the second one has.

**Comment by BrianCraig** [comment]
> About the warning: `react-i18next:: It seems you are still using the old wait option, you may migrate to the new useSuspense behaviour.`
> 
> I've also seen this in `webdev`, maybe we better upgrade to the newest version of `react-i18next` first? I hope it still supports class components.

Yes, but maybe i would stick with merging the tests and re-fixing them once we update the package, it may use Context instead of component global state.

**Comment by thisconnect** [comment]
ok with me, but could you skip the failing tests ? I think `test.skip` or `describe.skip` might work.

> Yes, but maybe i would stick with merging the tests and re-fixing them once we update the package,

**Comment by BrianCraig** [comment]
> ok with me, but could you skip the failing tests ? I think `test.skip` or `describe.skip` might work.
> 
> > Yes, but maybe i would stick with merging the tests and re-fixing them once we update the package,

There are no failing tests, they have warnings, if you wan't i could skip all that makes warning messages

**Comment by thisconnect** [comment]
Ah right, I just saw red error and automatically assumed test fail, didn't scroll down 😇 

> There are no failing tests, they have warnings, if you wan't i could skip all that makes warning messages

I wanted to ping @x1ddos for input as well.

File: frontends/web/.eslintrc.json [hunk 1]
```diff
@@ -9,6 +9,12 @@
     {
       "files": ["**/*.ts?(x)"],
       "rules": { }
+    },
+    {
+      "files": ["**/*.test.ts?(x)"],
+      "rules": {
+        "import/first": 0
+      }
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/.eslintrc.json:17]: lets keep as is and maybe do it when we rewrite the tests to testing/library

**Review by thisconnect — APPROVED** [review]
LGTM
This helps migration, but we shouldn't write new tests using Enzyme. Maybe even consider rewriting the tests after the migration is complete.

**Pull request merged**